src/Nethermind/Nethermind.JsonRpc.Test/Sockets/WebSocketTests.cs

```diff
@@ -0,0 +1,248 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Net;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.JsonRpc.WebSockets;
+using Nethermind.Logging;
+using Nethermind.Serialization.Json;
+using Nethermind.Sockets;
+using NUnit.Framework;
+
+namespace Nethermind.JsonRpc.Test.Sockets;
+
+class WebSocketsTests
+{
+    [Test]
+    [TestCase(2)]
+    [TestCase(10)]
+    [TestCase(50)]
+    public async Task Can_send_multiple_messages(int messageCount)
+    {
+        CancellationTokenSource cts = new();
+
+        Task<int> receiveMessages = OneShotServer(
+            "http://localhost:1337/",
+            async webSocket => await CountNumberOfMessages(webSocket, cts.Token)
+        );
+
+        Task<int> sendMessages = Task.Run(async () =>
+        {
+            using ClientWebSocket socket = new ClientWebSocket();
+            await socket.ConnectAsync(new Uri("ws://localhost:1337/"), CancellationToken.None);
+
+            using ISocketHandler handler = new WebSocketHandler(socket, NullLogManager.Instance);
+            using JsonRpcSocketsClient client = new JsonRpcSocketsClient(
+                clientName: "TestClient",
+                handler: handler,
+                endpointType: RpcEndpoint.Ws,
+                jsonRpcProcessor: null!,
+                jsonRpcService: null!,
+                jsonRpcLocalStats: new NullJsonRpcLocalStats(),
+                jsonSerializer: new EthereumJsonSerializer()
+            );
+            JsonRpcResult result = JsonRpcResult.Single(
+                new JsonRpcSuccessResponse
+                {
+                    MethodName = "mock", Id = "42", Result = new RandomObject(1000).Get()
+                },
+                default);
+
+            for (int i = 0; i < messageCount; i++)
+            {
+                await client.SendJsonRpcResult(result);
+                await Task.Delay(100);
```

Comment:
I would really prefer a different approach to ensure that the "server" has finished processing all messages instead of relying on delays.

Reply:
Use `ManualResetEventSlim`?

Reply:
Ended up adding some `[Retry(int)]` attributes